src/pybind/mgr/dashboard/frontend/src/app/shared/services/prometheus-alert.service.ts

```diff
@@ -57,9 +58,15 @@ export class PrometheusAlertService {
   }
 
   private handleAlerts(alerts: AlertmanagerAlert[]) {
+    this.silencedAlerts = 0;
     if (this.canAlertsBeNotified) {
       this.notifyOnAlertChanges(alerts, this.alerts);
     }
+    for (let index = 0; index < this.alerts.length; index++) {
```

Comment:
Instead of using the for-loop, use `reduce` from the lodash library. I don't think the reference to an explicit type is necessary, but I prefer to use explicit typing instead of implicit one.

```suggestion
    this.silencedAlerts = _.reduce<AlertmanagerAlert, number>(
      this.alerts,
      (result, alert) => {
        return (alert.status.state === 'suppressed') ? ++result : result;
      },
      0
    );
```

Reply:
Thanks @votdev . Modified accordingly